spidermon/contrib/scrapy/pipelines.py

```diff
@@ -94,8 +105,17 @@ def _load_jsonschema_validator(cls, schema):
                 "- an object path to a JSON string.\n"
                 "- a path to a JSON file."
             )
+
         return JSONSchemaValidator(schema)
 
+    @classmethod
+    def _load_cerberus_validator(cls, schema):
+        if isinstance(schema, six.string_types):
```

Comment:
Better to make these validations inside `__init__`.
Otherwise we need to carry them around whenever we want to use `CerberusValidator`

Reply:
The `__init__` of `pipelines.py`??
As per my understanding of slack messages, I thought we were keeping both the validations. 
`NotConfigured` on the outside, and SchemaError in the side, for the off chance if we ever need to pick the `CerberusValidator()` we can do that without worrying about validating our schemas.

Reply:
We need to check if the schema in `CerberusValidator`.
That's for sure, now, `CerberusValidator` shouldn't rase `NotConfigured` if it is invalid, maybe `SchemaError` is better.
So, here you can catch the given exception and raise `NotConfigured`.
The thing is that knowing if the given schema is valid is not something that `ItemValidationPipeline` should do, because the validation might vary among validators.

Reply:
> The thing is that knowing if the given schema is valid is not something that ItemValidationPipeline should do, because the validation might vary among validators.

Yes @ejulio this makes sense. `ItemValidationPipelines` shouldn't have one endall method to check the schema for all validators, rather custom checking could be the best option here.

Reply:
Shouldn't these validations go to `CerberusValidator.__init__`?

Reply:
Like we discussed, I guess. These validations are the same validations that are already present in `CerberusValidator` as the countermeasure to something that you said in regards to making code independent. 

As if someone would like to take our module then it should be independent and self-sustainable to use (Pardon my code vocabulary here :see_no_evil: )

Reply:
Or maybe I got it wrong, here's the `__init__` method of `CerberusValidator` class

```
   def __init__(self, schema, translator=None, use_default_translator=True):
        super(CerberusValidator, self).__init__(
            translator=translator, use_default_translator=use_default_translator
        )
        if isinstance(schema, Mapping):
            self._schema = schema
        else:
            raise SchemaError("Validation schema missing")
```

Reply:
Oooh, I see it now. I see you are referring to the `six.string_types` validation? 

Yes, that isn't there. And probably could be there. That's a good idea. Coming right up.

Reply:
Also, but if the validation is already in `CerberusValidator.__init__` why repeat it here?
Also, in `__init__` it is better to make the validations/assertions before any logic, in that case, calling `super`

Reply:
@ejulio do review this again. I spent about half the weekend to properly think and implement this. Let me explain what, why and where. 

- The `six.string_types` validation is actually for `URL's` and `paths`, that when `True` data can be sent inside the `get_schema_from()` method. This is placed outside in `pipelines.py` just because I wanted the things going inside the `CerberusValidator()` class to be just what we want and nothing else. This way testing would become much easier for us. As we need to test only schemas and not 15 other things. 

- Now, regarding our discussion of someone picking `CerberusValidator()` and using it somewhere else. Our purpose was to keep it independent. I think this achieves it as well. If you give `Mapping` type data in accordance with the Cerberus schema then this class would work. But, if you like to support URL and paths then the helper function `get_schema_from()` will have to be used. 

- The `Mapping` validation inside the `CerberusValidator()` stays there only. Because we need to check whatever schema that comes is in proper structure or not. However, we would need a lot of testing to see if this works. But, I am feeling good about this. Let me know what you think. I may have forgotten a thing or two since I am coming back to this discussion after a long time. Let me know.